examples/ex02_handwritten_digits_recognition.nim

```diff
@@ -33,26 +33,48 @@ let
   y_test = mnist.test_labels.astype(int)
 
 # Configuration of the neural network
-network ctx, DemoNet:
-  layers:
-    x:          Input([1, 28, 28])
-    cv1:        Conv2D(x.out_shape, 20, 5, 5)
-    mp1:        MaxPool2D(cv1.out_shape, (2,2), (0,0), (2,2))
-    cv2:        Conv2D(mp1.out_shape, 50, 5, 5)
+# network DemoNet:
+#   layers h, w:
+#     cv1:        Conv2D(@[1, h, w], 20, (5, 5))
+#     mp1:        Maxpool2D(cv1.out_shape, (2,2), (0,0), (2,2))
+#     cv2:        Conv2D(mp1.out_shape, 50, (5, 5))
+#     mp2:        MaxPool2D(cv2.out_shape, (2,2), (0,0), (2,2))
+#     fl:         Flatten(mp2.out_shape)
+#     hidden:     Linear(fl.out_shape[0], 500)
+#     classifier: Linear(500, 10)
+#   forward x:
+#     x.cv1.relu.mp1.cv2.relu.mp2.fl.hidden.relu.classifier
+
+network SomeConvNet:
+  layers h, w:
+    cv1:        Conv2D(@[1, h, w], 20, (5, 5))
+    mp1:        Maxpool2D(cv1.out_shape, (2,2), (0,0), (2,2))
+    cv2:        Conv2D(mp1.out_shape, 50, (5, 5))
     mp2:        MaxPool2D(cv2.out_shape, (2,2), (0,0), (2,2))
     fl:         Flatten(mp2.out_shape)
-    hidden:     Linear(fl.out_shape, 500)
-    classifier: Linear(500, 10)
   forward x:
-    x.cv1.relu.mp1.cv2.relu.mp2.fl.hidden.relu.classifier
+    x.cv1.relu.mp1.cv2.relu.mp2.fl
+
+proc out_shape*[T](self: SomeConvNet[T]): seq[int] =
+  self.fl.out_shape
+proc in_shape*[T](self: SomeConvNet[T]): seq[int] =
+  self.cv1.in_shape
```

Comment:
A section about "Composing models" can be added in the README to mention this convention